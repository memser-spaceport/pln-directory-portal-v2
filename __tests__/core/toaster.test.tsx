import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Toaster from '@/components/core/toaster';
import React from 'react';

// Section: Mock react-toastify ToastContainer to inspect props and rendering
jest.mock('react-toastify', () => ({
  ToastContainer: (props: any) => (
    <div data-testid="mock-toast-container" {...props} />
  ),
}));

/**
 * Test suite for Toaster component
 */
describe('Toaster', () => {
  it('renders the ToastContainer with correct props and classes', () => {
    const { getByTestId } = render(<Toaster />);
    const toastContainer = getByTestId('mock-toast-container');
    // Check that the ToastContainer is rendered
    expect(toastContainer).toBeInTheDocument();
    // Check all expected props are passed
    expect(toastContainer).toHaveAttribute('position', 'top-right');
    expect(toastContainer).toHaveAttribute('theme', 'dark');
    expect(toastContainer).toHaveAttribute('bodyclassname', 'toast__body');
    expect(toastContainer).toHaveAttribute('class', 'toast__class');
    expect(toastContainer).toHaveAttribute('toastclassname', 'toast__toaster');
    expect(toastContainer).toHaveAttribute('progressclassname', 'toast_progress');
  });

  it('matches snapshot for regression safety', () => {
    const { asFragment } = render(<Toaster />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Edge case: Rendering with extra props (future extensibility)
  it('renders without crashing if extra props are passed (future-proof)', () => {
    // @ts-expect-error: Testing future extensibility
    expect(() => render(<Toaster foo="bar" />)).not.toThrow();
  });
}); 