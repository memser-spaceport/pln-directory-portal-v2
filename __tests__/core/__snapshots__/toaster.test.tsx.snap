// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster matches snapshot for regression safety 1`] = `
<DocumentFragment>
  <div
    bodyclassname="toast__body"
    class="toast__class"
    data-testid="mock-toast-container"
    position="top-right"
    progressclassname="toast_progress"
    theme="dark"
    toastclassname="toast__toaster"
  />
</DocumentFragment>
`;
